exciting_environments/core_env.py

```diff
@@ -262,33 +248,38 @@ def default_reward_func(self, obs, action):
     @partial(jax.jit, static_argnums=0)
     @abstractmethod
     def generate_observation(self, states):
-        """Returns states."""
+        """Returns observation."""
         return states
```

Comment:
I think this should not return anything
```suggestion
        return
```
```suggestion
        pass
```
```suggestion
        raise NotImplementedError
```

are preferable IMO.